packages/data/src/index.js

```diff
@@ -18,7 +18,10 @@ export {
 	useRegistry,
 } from './components/registry-provider';
 export { default as useSelect } from './components/use-select';
-export { useDispatch } from './components/use-dispatch';
+export {
+	useDispatch,
+	useDispatchWithMap as __unstableUseDispatchWithMap,
```

Comment:
@youknowriad Just wanted to make you aware that I'm exporting `useDispatchWithMap` here (unstable API). It's needed for `RichText` because the format type needs access to `dispatch` to pick a store. I don't think there's any way around it, but maybe you have a better insight.

Reply:
> the format type needs access to dispatch to pick a store

Maybe, there's an alternative way to provide access to the hook in these types. Could you give a quick example about how we use dispatch in the formats?

Reply:
Sure:

https://github.com/WordPress/gutenberg/blob/30483cedaf9c8e16f62be85e241df06f7f745d93/packages/annotations/src/format/annotation.js#L147-L152